test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -112,7 +116,7 @@ func waitForFallbackDegradedCondition(t testing.TB, cs clientSet, waitPollTimeou
 	t.Helper()
 
 	t.Logf("Waiting for StaticPodFallbackRevisionDegraded condition, interval = 20s, timeout = %v", waitPollTimeout)
-	err := wait.Poll(20*time.Second, waitPollTimeout, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(context.Background(), 20*time.Second, waitPollTimeout, true, func(cxt context.Context) (bool, error) {
```

Comment:
we could also change the signature of the `waitForFallbackDegradedCondition` func and pass the ctx.

Reply:
Updated.